resources/conditions.ts

```diff
@@ -5,12 +5,37 @@
 import { Data } from '../types/data';
 import { Matches } from '../types/trigger';
 
+import {
+  StartsUsingParams,
+  AbilityParams,
+  AbilityFullParams,
+  HeadMarkerParams,
+  GainsEffectParams,
+  StatusEffectExplicitParams,
+  LosesEffectParams,
+  TetherParams,
+  WasDefeatedParams,
+  Regex,
+} from '../resources/regexes';
+
+type TargetableMatches = Matches<Regex<
+  StartsUsingParams |
+  AbilityParams |
+  AbilityFullParams |
+  HeadMarkerParams |
+  GainsEffectParams |
+  StatusEffectExplicitParams |
+  LosesEffectParams |
+  TetherParams |
+  WasDefeatedParams
+>>;
+
 export default {
-  targetIsYou(): (data: Data, matches: Matches) => boolean {
-    return (data: Data, matches: Matches) => data.me === matches.target;
+  targetIsYou(): (data: Data, matches: TargetableMatches) => boolean {
+    return (data: Data, matches: TargetableMatches) => data.me === matches?.target;
```

Comment:
Why does `matches?.target` need the null conditional operator? Shouldn't matches always be non-null?

Reply:
Just forgot to delete it(._.)

Reply:
Seems like it can be undefined sometimes?

Reply:
Sorry, could you explain to me how/when that could happen?

Reply:
for example, `Regexes.ability({ capture: false })`？

Reply:
Oh, right!